jedi/api/usages.py

```diff
@@ -39,8 +44,14 @@ def compare_array(definition_names):
         if isinstance(m, ModuleContext):
             for name_node in m.tree_node.used_names.get(search_name, []):
                 context = evaluator.create_context(m, name_node)
-                result = evaluator.goto(context, name_node)
-                if [c for c in compare_array(result) if c in compare_definitions]:
+                try:
+                    result = evaluator.goto(context, name_node)
+                except (NotImplementedError, RecursionError) as err:
+                    logging.getLogger(__name__).error(err)
+                    continue
```

Comment:
Yeah, I didn't mean to add that. I'll remove it.

BTW, I added that part a long time ago because the usages function was running into errors and then giving up. Catching the exception and continuing the loop allowed jedi to find other usages that didn't raise errors. I logged the error because I didn't want the errors to go unnoticed. What is an alternative way you would accept?